hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java

```diff
@@ -33,37 +33,38 @@
  */
 public class DefaultCyclicBuffer<E extends Sequenced> implements CyclicBuffer<E> {
```

Comment:
Can you explain from which threads this buffer is accessed, how and when? Why the need for such coarse synchronization?

Reply:
put a new assertion to have much info about failure and removed synchronization. This class is accessed by partition threads.